Hamming: panic

[Solution](https://exercism.io/mentor/solutions/bb47c6285ede4777847e43cddcd79730):

```go
//Package hamming contains function to calculate amount of character difference between two equal strings.
package hamming

//NotEqualLengthError is raised if strings supplied to Distance function have different length.
type NotEqualLengthError struct {}

func (error *NotEqualLengthError) Error() string {
	return "strings have different length"
}
//Distance calculates amount of character difference between two equal strings.
func Distance(a, b string) (int, error) {
	var lenA, lenB int
	lenA = len(a)
	lenB = len(b)

	if lenA != lenB {
		return -1, &NotEqualLengthError{}
	}

	if lenA == 0 {
		return 0, nil
	}

	var diff int

	for i := range a {
		if a[i] != b[i] {
			diff++
		}
	}

	return diff, nil

}
```

Panic:

```
./exalysis -watch
Watching for Exercism download links on the clipboard...
<-- Exalysis: /Users/john/git/bitfield/exercism/users/AlexRinn/go/hamming -->
Exercism package found: hamming
golint:          OK
gofmt:           FAIL
--- Current
+++ Formatted
@@ -5 +5 @@
-type NotEqualLengthError struct {}
+type NotEqualLengthError struct{}
@@ -9,0 +10 @@
+

go test:         OK
go test --race:  SKIPPED
go vet:          OK
golangci-lint:   OK
benchmarks:      SKIPPED
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/tehsphinx/exalysis/track/hamming.examReturnZeroValue(0xc0000be6e0, 0xc0002d2d80)
        /Users/john/git/bitfield/exalysis/track/hamming/hamming.go:35 +0x319
github.com/tehsphinx/exalysis/track/hamming.Suggest(0xc0000be6e0, 0xc0002d2d80)
        /Users/john/git/bitfield/exalysis/track/hamming/hamming.go:17 +0x7a
github.com/tehsphinx/exalysis.GetSuggestions(0xc000014700, 0x3b, 0xc0000bbea0, 0x1, 0x1, 0x4e)
        /Users/john/git/bitfield/exalysis/suggestion.go:61 +0x503
main.watchClipboard()
        /Users/john/git/bitfield/exalysis/cmd/exalysis/main.go:64 +0x23e
main.main()
        /Users/john/git/bitfield/exalysis/cmd/exalysis/main.go:31 +0x26a
```

--- a/track/hamming/hamming.go
+++ b/track/hamming/hamming.go
@@ -30,9 +30,9 @@ var exFuncs = []extypes.SuggestionFunc{
 }
 
 func examReturnZeroValue(pkg *astrav.Package, r *extypes.Response) {
-	returns := pkg.FindByNodeType(astrav.NodeTypeReturnStmt)
+	returns := pkg.FindFirstByName("Distance").FindByNodeType(astrav.NodeTypeReturnStmt)
 	for _, ret := range returns {
-		if ret.Children()[1].ValueType().String() != "error" {
+		if ret.Children()[1].ValueType().String() == "nil" {
 			continue
 		}
 		switch node := ret.Children()[0].(type) {

--- a/track/hamming/hamming_test.go
+++ b/track/hamming/hamming_test.go
@@ -37,6 +37,7 @@ var suggestTests = []struct {
 	{path: "./solutions/6", suggestion: tpl.ReturnZeroValue, expected: true},
 	{path: "./solutions/7", suggestion: tpl.NoErrMsg, expected: true},
 	{path: "./solutions/7", suggestion: tpl.ReturnZeroValue, expected: true},
+	{path: "./solutions/8", suggestion: tpl.ReturnZeroValue, expected: true},
 }
 
 func Test_Suggest(t *testing.T) {

--- a/track/hamming/solutions/8/solution.go
+++ b/track/hamming/solutions/8/solution.go
@@ -1,0 +1,35 @@
+// Package hamming contains function to calculate amount of character difference between two equal strings.
+package hamming
+
+// NotEqualLengthError is raised if strings supplied to Distance function have different length.
+type NotEqualLengthError struct{}
+
+func (error *NotEqualLengthError) Error() string {
+	return "strings have different length"
+}
+
+// Distance calculates amount of character difference between two equal strings.
+func Distance(a, b string) (int, error) {
+	var lenA, lenB int
+	lenA = len(a)
+	lenB = len(b)
+
+	if lenA != lenB {
+		return -1, &NotEqualLengthError{}
+	}
+
+	if lenA == 0 {
+		return 0, nil
+	}
+
+	var diff int
+
+	for i := range a {
+		if a[i] != b[i] {
+			diff++
+		}
+	}
+
+	return diff, nil
+
+}

